docs/sources/tempo/configuration/_index.md

```diff
@@ -707,6 +710,34 @@ query_frontend:
 
 ```
 
+### Limit query size to improve performance and stability
+
+Querying large tracing data presents several challenges.
+Span sets with large number of spans impact query performance and stability.
+In a similar manner, excessive queries result size can also negatively impact query performance.
+
+#### Limit the spans per spanset
+
+You can set the maximum spans per spanset by setting `max_spans_per_span_set` for the query-frontend.
+The default value is 100. Tempo and Grafana Enterprise Traces users can set this option in their configuration file.
+
+In Grafana or Grafana Cloud, you can set an equivalent option by using the **Span Limit** field in the [TraceQL query editor](https://grafana.com/docs/grafana-cloud/connect-externally-hosted/data-sources/tempo/query-editor/) in Grafana Explore.
```

Comment:
My understanding of https://github.com/grafana/tempo/pull/4383 is that the **Span Limit** from Grafana is not an equivalent of `max_spans_per_span_set`. Rather, with this change the maximum value that users can set for the **Span Limit** value (or the `spss` query) is now `max_spans_per_span_set` (which by default is 100).

@joe-elliott @mdisibio Can you chime in please?

Reply:
We can easily rework this. What about this? 
```suggestion
In Grafana or Grafana Cloud, you can use the **Span Limit** field in the [TraceQL query editor](https://grafana.com/docs/grafana-cloud/connect-externally-hosted/data-sources/tempo/query-editor/) in Grafana Explore.
```

Reply:
> My understanding of https://github.com/grafana/tempo/pull/4383 is that the Span Limit from Grafana is not an equivalent of max_spans_per_span_set. Rather, with this change the maximum value that users can set for the Span Limit value (or the spss query) is now max_spans_per_span_set (which by default is 100).

Correct 👍

Reply:
> What if we remove "equivalent option" and instead use this?

@knylander-grafana makes sense

I propose below a draft, but feel free to adapt it as needed.

```suggestion
You can set the maximum spans per span set that can be requested by a TraceQL query by setting `max_spans_per_span_set` for the query-frontend. The default value is 100.

If your query requests a number of spans per span set above this maximum, it will return an error.

In Grafana or Grafana Cloud, the value of **Span Limit** field in the [TraceQL query editor](https://grafana.com/docs/grafana-cloud/connect-externally-hosted/data-sources/tempo/query-editor/) in Grafana Explore cannot exceed the configured maximum.
```